designs/static-capacity.md

```diff
@@ -0,0 +1,141 @@
+# Karpenter - Static Capacity
+
+## Background
+Karpenter currently operates as a dynamic cluster autoscaler, automatically adjusting node counts based on pending pod demand. However, several important use cases require maintaining a fixed set of nodes:
+
+1. Performance-critical applications where just-in-time provisioning latency is unacceptable
+2. Workloads that need guaranteed capacity regardless of pod demand  
+3. Scenarios where fixed node counts are preferred
+
+Currently, users attempt to achieve static capacity through workarounds:
+- Running placeholder pods to maintain minimum capacity
+- Using separate node management tools alongside Karpenter (MNG/Fargate/Self-managed)
+- Configuring complex provisioner requirements to approximate static behavior
+
+## Proposal
+Extend the existing NodePool resource to support static provisioning capabilities by adding new fields:
+
+```yaml
+spec:
+  # Makes NodePool static when specified
+  replicas: 5  # Number of nodes to maintain
+```
+
+Key aspects:
+1. Static NodePools maintain fixed node count regardless of pod demand
```

Comment:
Can we define what you mean by static and dynamic NodePool to make it clear for new readers on the doc?

Reply:
Its there in the yaml, will add explicitly